### DISCUSSION TITLE
How to share common libraries across multiple console apps

### DISCUSSION THREAD

**Discussion by askids** [nulastudio/NetCoreBeauty#43] [Q&A] [answered]
hi,

I currently have a .Net FW solution that has some 9 projects each generating a console app. I am migrating the app to .Net Core. On a trial basis, I want to package the .Net Core apps and deploy on the same server, without installing .Net 6 as we will eventually run these apps on separate set of servers. Now, if I try to have self contained single file deployment, each project is generating large files (~90MB with compression, trimming enabled) which will blow up my installer size. So using NetBeauty will I be able to publish this with overall smaller size, where the apps can shared the common libraries. If so, can you please guide me on how to achieve this? 

FYI, we are using MSBuild 17 and doing the build using the solution file via Jenkins pipeline and then use wix to package the required files into a MSI and publish the resulting MSI artifact.

Also, does Nbeauty CLI has to be installed on the build server as well or is having package reference sufficient?

Thanks!

**Comment by liesauer** [comment, marked as answer]
you can have a look at this example, https://github.com/nulastudio/NetBeauty2/tree/master/NetBeautyTest/SharedRuntimeTest

the key configs are
```xml
<BeautySharedRuntimeMode>True</BeautySharedRuntimeMode>
<BeautyLibsDir Condition="$(BeautySharedRuntimeMode) == 'True'">../runtimes</BeautyLibsDir>
```

so the final structure will be something likes this
```text
├── runtimes
│   └── tons of runtime dlls and dependencies
├── app1
│   ├── hostfxr.dll
│   ├── nbloader.dll
│   ├── app1.deps.json
│   ├── app1.dll
│   ├── app1.exe
│   ├── app1.runtimeconfig.json
│   └── ...
├── app2
│   ├── hostfxr.dll
│   ├── nbloader.dll
│   ├── app2.deps.json
│   ├── app2.dll
│   ├── app2.exe
│   ├── app2.runtimeconfig.json
│   └── ...
└── more apps
```

and all you need to do is referencing the NetBeauty nuget package and some config, `nbeauty` cli is unneeded, it's for tidying up a published application.

currently, NetBeauty doesn't support tidying up multiple apps into one folder, so the following structure is impossible for now
```text
├── runtimes
│   └── tons of runtime dlls and dependencies
└── apps
    ├── hostfxr.dll
    ├── nbloader.dll
    ├── app1.deps.json
    ├── app1.dll
    ├── app1.exe
    ├── app1.runtimeconfig.json
    ├── app2.deps.json
    ├── app2.dll
    ├── app2.exe
    ├── app2.runtimeconfig.json
    └── ...
```
